lte/gateway/c/oai/tasks/sgw_s8/sgw_s8_handlers.c

```diff
@@ -1033,3 +1033,39 @@ static void sgw_s8_send_failed_delete_session_response(
   }
   OAILOG_FUNC_OUT(LOG_SGW_S8);
 }
+
+/* From GPP TS 23.401 version 11.11.0 Release 11, section 5.3.5 S1 release
+   procedure: The S-GW releases all eNodeB related information (address and
+   TEIDs) for the UE and responds with a Release Access Bearers Response message
+   to the MME. Other elements of the UE's S-GW context are not affected. The
+   S-GW retains the S1-U configuration that the S-GW allocated for the UE's
+   bearers. The S-GW starts buffering downlink packets received for the UE and
+   initiating the "Network Triggered Service Request" procedure, described in
+   clause 5.3.4.3, if downlink packets arrive for the UE.
+*/
+//------------------------------------------------------------------------------
+void sgw_s8_handle_release_access_bearers_request(
+    const itti_s11_release_access_bearers_request_t* const
+        release_access_bearers_req_pP,
+    imsi64_t imsi64) {
+  OAILOG_FUNC_IN(LOG_SGW_S8);
+  OAILOG_DEBUG_UE(
+      LOG_SGW_S8, imsi64,
+      "Release Access Bearer Request Received in SGW_S8 task \n");
+
+  sgw_eps_bearer_context_information_t* sgw_context_p =
+      sgw_get_sgw_eps_bearer_context(release_access_bearers_req_pP->teid);
+  if (sgw_context_p) {
+    sgw_send_release_access_bearer_response(
+        LOG_SGW_S8, imsi64, REQUEST_ACCEPTED, release_access_bearers_req_pP,
+        sgw_context_p->mme_teid_S11);
+    sgw_process_release_access_bearer_request(
```

Comment:
This won't work because instead of `del_s8_tunnel`, `del_tunnel` is used. S8task and SPGW tasks utilize different ports to add flow match rules for DL.

Reply:
Will not Del_s8_tunnel remove both tunnels (s1ap and s8)?

Don't we have to just release s1ap tunnel when we go to idle? That is why I think here we need to use del_runne

Reply:
I think the reason why we have a separate API for s8 tunnel deletion is due to the explicit match on port plus the flow match. If there were no port match, deleting all flow rules for the UE (for default and dedicated) would have been taken care of by specifying the UE IP address. @pshelar could you cross-check my comment here?

Reply:
During testing below is my observation:
Del_s8_tunnel() deletes both tunnels
Del_tunnel() with s1-u tunnel info, is retaining s1-u tunnel info and deleting s8-u tunnel info
Del_tunnel() with s8-u tunnel info, is retaining both s1-u and s8-u tunnels.

Reply:
We are using del_tunnel() to avoid deleting both UL and DL tunnels. I think for idle mode we need to keep DL tunnel flows, del_tunnel() works by deleting only UL S8 flows.
I can add separate API for this specifically for S8 UL tunnel delete.

Reply:
in idle mode we should not have forwarding rules to/from eNB and we should have the paging rule. The tunnels between AGW and PGW (i.e., S8 tunnels) should be intact during idle mode.

Reply:
S8 flows programmed like this:
UL: eNB-tunnel -> output to PGW 
DL: PGW-tunnel -> output to eNB tunnel.

this way four rules are actually merged in two rules in datapath.
 
so it is not possible to delete rules to/from eNB and keep tunnel to PGW.

Reply:
so that means we can't support IDLE mode ?

Reply:
it means we need to install a new rule for paging support:
UL: we can keep that rule, there is no harm...
DL: PGW-tunnel -> packet in to controller

Reply:
@rsarwad  for now we will use `delete_s8_tunnel` to remove the tunnels when the UE goes IDLE, and then create them back again when the UE goes connected mode